fix: timezone parsing issues for calendars

curiousgeorgios opened the pull request:
## What does this PR do?

<!-- Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change. -->

- After reviewing current issues, there are a number of issues raised in relation to calendar discrepancies, which this PR is likely to solve. 
- I couldn't find a specific issue in Linear calling this timezone bug out. I noticed and troubleshot this bug myself as I noticed a discrepancy between the timezone in my ICS calendar and what Cal was showing.

## Visual Demo (For contributors especially)

Below is a troubleshooting event that I setup on my ICS calendar provider
```text
BEGIN:VEVENT
UID:`REDACTED`
DTSTAMP:20250601T105202Z
SUMMARY:11-12 on 4 June - +2 Timezone
DTSTART;TZID=Europe/Brussels:20250604T110000
DTEND;TZID=Europe/Brussels:20250604T120000
SEQUENCE:0
STATUS:CONFIRMED
END:VEVENT
```

And this event is showing as 0300 to 0400 (in the same timezone settings in cal) 
<img width="791" alt="image" src="https://github.com/user-attachments/assets/243ebf15-6310-46d8-9bb3-f369dc366d8b" />


#### Video Demo (if applicable):

Not applicable

#### Image Demo (if applicable):

See screenshots above

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox. **N/A**
- [ ] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

<!-- Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration. Write details that help to start the tests -->

- Are there environment variables that should be set?
- What are the minimal test data to have?
- What is expected (happy path) to have (input and output)?
- Any other important info that could help to test that PR

## Checklist

<!-- Remove bullet points below that don't apply to you -->


    
<!-- This is an auto-generated description by cubic. -->
---

## Summary by cubic
Fixed calendar timezone parsing to correctly handle TZID values from ICS files, preventing event time discrepancies.

<!-- End of auto-generated description by cubic. -->

curiousgeorgios committed:
fix timezone parsing issues for calendars

Devanshusharma2005 reviewed (commented):
Address the cubic suggestions.

curiousgeorgios committed:
fix: maintain backward compatibility with other timezone specification methods

accepting the cubic-dev-ai as implement backward compatibility with different event ICS event types

Co-authored-by: cubic-dev-ai[bot] <191113872+cubic-dev-ai[bot]@users.noreply.github.com>

curiousgeorgios commented:
> Address the cubic suggestions.

thanks @Devanshusharma2005 - updated.

Devanshusharma2005 commented:
@curiousgeorgios getting typecheck issues.

Devanshusharma2005 commented:
marking this pr as closed due to inactivity. Feel free to ping me up if you want to continue to work on this.

anikdhabal committed:
Update Cubic comment

anikdhabal committed:
Merge branch 'main' into main

anikdhabal reviewed (dismissed):
Looking good

curiousgeorgios commented:
> Looking good

thank you brother

curiousgeorgios commented:
> @curiousgeorgios could you pls also attach the after image

yep will do that 👌

kart1ka commented:
@curiousgeorgios A type check seems to be failing. Can you fix that?

curiousgeorgios committed:
Fix timezone handling in ICS feed calendar service

- Fix VTIMEZONE selection to find matching timezone instead of using first one
- Improve TZID parameter extraction from DTSTART property using jCal array
- Fix precedence: DTSTART TZID > standalone TZID > UTC > fallback
- Resolves issue where Europe/Brussels events showed wrong times in Europe/Paris

curiousgeorgios commented:
@anikdhabal grateful for your review to try and get this merged

kart1ka committed:
Merge branch 'main' into main

anikdhabal committed:
remove comments

anikdhabal committed:
remove comments

anikdhabal reviewed (approved):
👍

anikdhabal committed:
Merge branch 'main' into main

anikdhabal committed:
Merge branch 'main' into main

anikdhabal committed:
Merge branch 'main' into main
